### PR TITLE
Break down sensor combined

### PR DESCRIPTION
**Pull request by bkueng**
As discussed in https://github.com/PX4/Firmware/issues/4861

- This reduces the size of `sensor_combined` by a factor of 10 (!) from 790 to 72 bytes.
- move voting into sensors module. This means that now everyone using `sensor_combined` gets voted data, including ekf2. It also adds voting for the baro.
- use relative timestamps, necessary for the new logger. Logging `sensor_combined` at full rate now only needs 18KB/s.
- Memory savings: when using the `q` estimator, we get almost 2kB of additional RAM, with ekf2 it's roughly 0.6kB. This is because ekf2 did not do voting, and voting itself is quite heavy, it requires 1.5kB.

This introduces a lot of changes at central places, affecting all estimators. Nothing complex, but still, it needs careful review and testing all of the estimators.

I still plan to break up the sensors class into several smaller classes to make things easier to maintain.

### CODE REVIEW

**Comment by bkueng** [comment]
This was flight tested with inav and q estimators.

**Comment by LorenzMeier** [comment]
@bkueng What is the value of renaming sensor_accel to sensor_accel_raw? Unless there is clear value on top of just naming I would like to avoid that change.

**Comment by bkueng** [comment]
It makes it more clear that it's the raw sensor data from the driver. Initially I planned to split up sensor_combined into several topics which would have gotten the name `sensor_accel` etc. However as sensor_combined is now really small, it's not necessary to split it, so the topic rename is not strictly necessary either.

**Comment by LorenzMeier** [comment]
I don't think that the rename at this point adds significant clarity and I know that a lot of people are using the topics directly. So I think that's a change we shouldn't make at this point.

**Comment by bkueng** [comment]
Fine with me. I'll drop it when rebasing. Do you plan to merge it now?

**Comment by LorenzMeier** [comment]
Yes, I think it makes sense to get these in.

**Comment by bkueng** [comment]
rebased & dropped sensor topics renaming

File: src/drivers/frsky_telemetry/frsky_data.c [hunk 1]
```diff
@@ -204,20 +204,22 @@ void frsky_update_topics()
...
-			roundf(sensor_combined->accelerometer_m_s2[0] * 1000.0f));
-	frsky_send_data(uart, FRSKY_ID_ACCEL_Y,
-			roundf(sensor_combined->accelerometer_m_s2[1] * 1000.0f));
-	frsky_send_data(uart, FRSKY_ID_ACCEL_Z,
-			roundf(sensor_combined->accelerometer_m_s2[2] * 1000.0f));
+	float acceleration[3];
+	float accel_dt = sensor_combined->accelerometer_integral_dt / 1.e6f;
+	acceleration[0] = sensor_combined->accelerometer_integral_m_s[0] / accel_dt;
+	acceleration[1] = sensor_combined->accelerometer_integral_m_s[1] / accel_dt;
+	acceleration[2] = sensor_combined->accelerometer_integral_m_s[2] / accel_dt;
```
Comment (julianoes) [inline src/drivers/frsky_telemetry/frsky_data.c]: Having to add the division through dt everywhere seems painful. Of course, you've done it already, I'm just thinking of devs that will get confused. Are we sure we want to do this?

File: msg/sensor_combined.msg [hunk 2]
```diff
@@ -1,73 +1,24 @@
...
-uint64[3] accelerometer_integral_dt		# delta time for accel integral in us
-int16[3] accelerometer_mode			# Accelerometer measurement mode
-float32[3] accelerometer_range_m_s2		# Accelerometer measurement range in m/s^2
-uint64[3] accelerometer_timestamp	# Accelerometer timestamp
-uint32[3] accelerometer_priority	# Sensor priority
-uint32[3] accelerometer_errcount	# Error counter for accel 0
-float32[3] accelerometer_temp		# Temperature of accel 0
+# gyro timstamp is equal to the timestamp of the message
+float32[3] gyro_integral_rad		# delta angle in the NED body frame in rad in the integration time frame
+uint32 gyro_integral_dt			# delta time for gyro integral in us
```
Comment (julianoes) [inline msg/sensor_combined.msg]: I would argue this should be a `float32` in seconds. I'm not completely happy that you need to do `/ 1.e6f;` everywhere below.

**Comment by bkueng** [comment]
@julianoes I agree in both cases. I discussed this with @tumbili and he prefered the integral because the estimators use it. But when looking at the overall usage, most places need the value, not the integral. Personally I also prefer the values.
Plus when thinking about the new logger, the value can be plotted directly.

**Comment by julianoes** [comment]
I like the changes overall but I would vote to change the sensor_combined struct and make the API as simple as possible for devs.

Therefore, I'd publish the accel/gyro/mag values without integral plus timestamp. If an estimator really wants integrals, it should just multiply the values with delta_t. I think this would make it cleaner and also better performance-wise because it would lead to less divisions.
I'm just not sold on the idea of integrals instead of intuitive values.

**Comment by bkueng** [comment]
just to be clear, you mean like this? sensor_combined.msg:
```
float32[3] gyro_rad
float gyro_integral_dt # in seconds

int32 accelerometer_timestamp_relative
float32[3] accelerometer_m_s
float accelerometer_integral_dt	

int32 magnetometer_timestamp_relative
float32[3] magnetometer_ga	

int32 baro_timestamp_relative
float32 baro_alt_meter		
float32 baro_temp_celcius
```

**Comment by julianoes** [comment]
@bkueng yes that's what I would suggest.

**Comment by LorenzMeier** [comment]
Looks sensible to me.

**Comment by bkueng** [comment]
ok if no one objects, I'll change this.

**Comment by bkueng** [comment]
refactored to the above format

**Comment by LorenzMeier** [comment]
Rebased and applied.

**Pull request closed without merging**